zkevm-circuits/src/pi_circuit.rs

```diff
@@ -1486,14 +1486,27 @@ impl<F: Field> PiCircuit<F> {
         block: &Block<F>,
     ) -> Self {
         let chain_id = block.chain_id;
+        let next_state_root = block
+            .context
+            .ctxs
+            .last_key_value()
+            .map(|(_, blk)| blk.eth_block.state_root)
+            .unwrap_or(H256(block.prev_state_root.to_be_bytes()));
+        // sanity check
+        assert_eq!(
+            block.mpt_updates.new_root().to_be_bytes(),
+            next_state_root.to_fixed_bytes()
+        );
```

Comment:
this should be log::error.

The above 2 assert_eq make sense since they are "mis-config".  We should stop as soon as possible and re-config and re-run.

But for roots mismatch, it is "normal err".   We should not assert_eq directly.